### DISCUSSION TITLE
What particle list is selected for Particle refinement

### DISCUSSION THREAD

**Discussion by JuhaHuiskonen** [nextpyp/.github#61] [Q&A] [answered]
I have run particle detection (pyp-eval) a few times now with different parameters. After each run, I have copied `Auto particles` to a new set.  How can I define the set of particles for refinement?

**Comment by abartesaghi** [comment]
You specify this in the particle refinement block downstream. There should be a drop down menu at the top of the parameters page where you can specify the particle list you want to use for refinement. If you want to process different lists of particles, you will need to create a refinement block for each of them.

**Comment by JuhaHuiskonen** [comment]
There is a dropdown menu in the Pre-processing block.  I select a list of particles there, but the number of particles is not updated.

**Reply by abartesaghi** [reply]
Sorry, my answer was confusing. You need to select the list of particles from the dropdown menu in the Pre-processing block (not the Refinement block as I said above), and you also need to select the option "Manual" for the `Detection method` in the Particle detection tab.

**Reply by JuhaHuiskonen** [reply]
I am selecting the the list of particles from the dropdown menu in the Pre-processing block and also he option "Manual (list of positions required)" for the Detection method in the Particle detection tab. When I save this modified block, the number of particles is not updated. It is shown as 0.  

<img width="295" alt="Screenshot 2024-02-09 at 8 33 04" src="https://github.com/nextpyp/.github/assets/20352352/0978d68d-941e-4879-a9be-693870c49e63">

**Reply by abartesaghi** [reply, marked as answer]
Yes, this is expected. The particle number will be updated once the block finishes running

**Reply by JuhaHuiskonen** [reply]
Do I need to rerun the Pre-processing block every time I change the particle list?

**Reply by abartesaghi** [reply]
Yes, but these will be lightweight jobs that should run very quickly.

**Reply by JuhaHuiskonen** [reply]
When jobs are submitted to SLURM, it might take a long time for them to start... Lightweight jobs could be run on the login node.

**Reply by abartesaghi** [reply]
Yes, they are lightweight but they still need to be submitted to slurm. As a general rule, nextpyp doesn't run any data processing operations on the login node. One reason for this is portability, since we've seen some setups where apptainer processes are not allowed on the login node. if jobs are slow to run because of the scheduler, you can try bundling them together using the `Bundle size` parameter in the resources tab. This will allow you to process multiple tilt series serially within the same slurm job and can help reduce wait times.

**Comment by JuhaHuiskonen** [comment]
I rerun the pre-processing block now. The particle list with 79 particles in selected in the dropdown menu.  The run completes (and it is indeed fast) but the particle count is not updated.

**Reply by abartesaghi** [reply]
We can't reproduce this, do you see any errors in the merge job?

**Reply by JuhaHuiskonen** [reply]
There are no errors printed. The number of autopicked particles is displayed correctly.  But if I simply pick manually, and rerun the block, it still shows 0 particles.
